code/game/objects/effects/anomalies/anomaly_bleed.dm

```diff
@@ -0,0 +1,60 @@
+/obj/effect/anomaly/blood
+	name = "blood anomaly"
+	icon_state = "blood_anomaly"
+	density = TRUE
+	aSignal = /obj/item/assembly/signaler/anomaly/blood
+	var/sucking = FALSE
+
+/obj/effect/anomaly/blood/anomalyEffect(delta_time)
+	if (sucking)
+		return
+	..()
+	if (DT_PROB(20, delta_time))
+		AddComponent(/datum/component/pellet_cloud, projectile_type=/obj/projectile/bullet/shrapnel/bleed, magnitude=3)
+		playsound(src, 'sound/weapons/shrapnel.ogg', 70, TRUE)
+	if (DT_PROB(10, delta_time))
+		for (var/mob/living/carbon/bleed_target in view(6, src))
+			if (!bleed_target.resting)
+				continue
+			if (do_teleport(src, get_turf(bleed_target), channel = TELEPORT_CHANNEL_BLUESPACE))
+				bleed_target.Stun(3 SECONDS)
+				INVOKE_ASYNC(src, PROC_REF(suck_blood))
+			break
+
+/obj/effect/anomaly/blood/detonate()
+	STOP_PROCESSING(SSobj, src)
+	// Needs to sleep since this gets instantly deleted as soon as the proc ends
+	for (var/mob/living/carbon/human/player in GLOB.player_list)
+		if (!is_station_level(player.z))
+			continue
+		var/turf/player_loc = get_turf(player)
+		var/list/nearby_turfs = RANGE_TURFS(4, player_loc)
+		shuffle_inplace(nearby_turfs)
+		var/turf/target = locate(/turf/open) in nearby_turfs
+		if (!target)
+			continue
+		// Blocked by the bluespace anchor
+		if (!do_teleport(src, target, channel = TELEPORT_CHANNEL_BLUESPACE))
+			return
```

Comment:
This logic should be more accurate.
* Anom [Blocked] -> Mob [Open]: This should be early return at the start of the proc because the anomaly power will not be sent to anywhere.
* Anom [Open] -> Mob [Blocked]: This is where this condition is applied, and this means only the loc of the mob is blocked. This means the anomaly can affect other areas.

Reply:
I don't quite understand this. If the anomaly is blocked by a bluespace anchor when it attempted to teleport, then `detonate()` will return which deletes it. I don't want the anomaly to continue attacking people if it gets blocked by the bluespace anchor.

All logic for the bluespace anchor is handled by `do_teleport`. The bluespace anchor blocks teleportation in and out regardless and is a standardised behaviour.

Reply:
Let's me elaborate it more.
* If departing (anomly) turf is anchored, should the anomaly be able to _do_ teleport?
* If arrival (target) turf is anchored, should the anomaly be able to teleport _to_ there?

First case would be where `detonate()` is fully suppressed due to bluespace anchor.
Second case would be where `detonate()` is partially effective to people who aren't under bluespace anchor.
but the current code you wrote is "suppress `detonate()` when there are people who are under bluespace anchor` which is a bit not consistent. This means `detonate()` will hurt people until it gets a person who's protected by bluespace anchor meanwhile it will skip all other people who aren't under bluespace anchor.

Reply:
Yes, it will fire at people until it gets blocked by a bluespace anchor, whether teleporting into or out of a bluespace anchor. This is intentional behaviour and the bluespace anchoring blocking is handled by do_teleport and is consistent across all applications of it.

Reply:
So, it is intended that:
* when there are 10 people
* index 5 person is under bluespace anchor
* anomaly will attack index 1-4 people
* anomaly will stop attacking index 5 person (because of bluespace anchor)
* index 6-10 people will not be attacked

right? it sounds not consistent to me, but if it is meant to be intended, then okay.

Reply:
Yes